src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+    "task": "Finish the report"
+  }
+}
+```
+
+### Update a task
+To update an existing task, use the following cURL command, replacing `<task_id>` with the id of the task, and `<updated_data>` with the JSON data you want to update:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '<updated_data>' http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X PUT -H "Content-Type: application/json" -d '{"completed":true}' http://127.0.0.1:5000/tasks/3
+```
+The `curl` command sends an HTTP PUT request with JSON data to `http://127.0.0.1:5000/tasks/3`, updating the task with ID `3` to mark it as completed.
+
+Expected Result:
+```json
+{
+  "task": {
+    "completed": true,
+    "id": 3,
+    "task": "Finish the report"
+  }
+}
+```
+
+### Delete a task
+To delete a task, use the following cURL command, replacing `<task_id>` with the id of the task you want to delete:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/<task_id>
+```
+Example:
+```bash
+curl -X DELETE http://127.0.0.1:5000/tasks/3
+```
+The `curl` command sends an HTTP DELETE request to `http://127.0.0.1:5000/tasks/3`, deleting the task with ID `3`.
+
+Expected Result:
+```json
+{
+  "result": "Task deleted"
+}
+```
+
+Let's try to retrieve a task that doesn't exist in the list of tasks. For instance, we just deleted a task with id = 3. Now, let's try to retrieve the task with id = 3 and see the result.
+```bash
+curl http://127.0.0.1:5000/tasks/3
+```
+Expected result:
+```json
+{
+  "error": "Task not found"
+}
+```
+We see that the JSON response returns the error `Task not found`, which indicates that the task doesn't exist in the list of tasks.
+
+## Troubleshoot
+Now, let's explore scenarios where API requests might fail. API requests can fail due to various reasons, including sending invalid requests. This could happen if an incorrect endpoint is used, the JSON data is invalid, or if the request is made with an unsupported HTTP method. Let's examine some instances where API requests might fail.
+
+**1. Invalid endpoint**:
+With the REST API that we built, we have the following valid API endpoints:
+- `GET /tasks`: Retrieves a list of all available tasks
+- `GET /tasks/<int:task_id>`: Retrieves a specific task by its ID
+- `POST /tasks`: Creates a new task
+- `PUT /tasks/<int:task_id>`: Updates a specific task by its ID
+- `DELETE /tasks/<int:task_id>`: Deletes a specific task by its ID
+
+If we send an API request to any of the endpoint rather then the listed endpoints, the request will fail with an stauts code `404`. For instance, let's try to get all the tasks with following cURL command:
+```bash
+curl http://127.0.0.1:5000/task
+```
+Here, we have used `task` instead of `tasks` in our endpoint, which makes our endpoint invalid. With this request we get the following error with `404` status code.
+
+Expected Result:
+```console
+The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
```

Comment:
```suggestion
The requested URL was not found on the server. If you entered the URL manually, please check your spelling and try again.
```
not sure if the error message has a comma... grammatically it would be better to set a comma, but if that's the actual error message, don't modify it.

Reply:
That is the actual error message.